In [1]:
import numpy as np
import pandas as pd

In [2]:
df = pd.read_csv('./datasets/recipe_utf8.csv')
df

Unnamed: 0  ROW_NUM  RECIPE_ID RECIPE_NM_KO  \
0             0        1          1        나물비빔밥   
1             1        2          2          오곡밥   
2             2        3          3          잡채밥   
3             3        4          4         콩나물밥   
4             4        5          5           약식   
..          ...      ...        ...          ...   
532         532      533     180330      식빵고구마파이   
533         533      534     180344   카레토마토달걀볶음밥   
534         534      535     180363      쪽파 새우강회   
535         535      536     195428         누드김밥   
536         536      537     195453      콩비지동그랑땡   

                                                 SUMRY  NATION_CODE NATION_NM  \
0                      육수로 지은 밥에 야채를 듬뿍 넣은 영양만점 나물비빔밥!      3020001        한식   
1                       정월대보름에 먹던 오곡밥! 영양을 한그릇에 담았습니다.      3020001        한식   
2              잡채밥 한 그릇이면 오늘 저녁 끝! 입 맛 없을 때 먹으면 그만이지요~      3020004        중국   
3           다이어트에 으뜸인 콩나물밥. 밥 물 넣을때 평소보다 적게 넣는거 잊지마세요!      3020001        한식   
4         집에서도 쉽게 만들어 맛있게 먹을 수 있답니다. 어려워 마시고 만들어 보세요~!      3020001        한식   
..                                                 ...          ...       ...   
532                   식빵을 이용한 간단 고구마파이! 아이들 영양 간식으로 딱!      3020002        서양   
533                 토마토와 달걀을 넣은 볶음밥에 카레소스를 두른 영양만점 볶음밥      3020009        퓨전   
534  새우에 쪽파를 감아 간단하게 만들지만 새우와 쪽파의 달콤한 맛과 새콤한 초고추장의 ...      3020009        퓨전   
535               건강한 재료로 가득담아 말은 누드김밥! 고기가 없어도 맛있어요!!      3020009        퓨전   
536                       두부대신 콩비지를 넣어 만든 동그랑땡 맛도 좋아요!      3020001        한식   

     TY_CODE      TY_NM COOKING_TIME  CALORIE  QNT LEVEL_NM IRDNT_CODE  \
0    3010001          밥          60분  580Kcal  4인분       보통         곡류   
1    3010001          밥          60분  338Kcal  4인분       보통         곡류   
2    3010001          밥          30분  520Kcal  4인분       보통         곡류   
3    3010001          밥          40분  401Kcal  4인분     초보환영         곡류   
4    3010013       떡/한과          60분  259Kcal  4인분       보통         곡류   
..       ...        ...          ...      ...  ...      ...        ...   
532  3010014       빵/과자          30분    0Kcal  2인분     초보환영        NaN   
533  3010001          밥          30분    0Kcal  1인분     초보환영        NaN   
534  3010007  나물/생채/샐러드          20분    0Kcal  2인분     초보환영        NaN   
535  3010001          밥          20분    0Kcal  3인분       보통        NaN   
536  3010018         부침          30분    0Kcal  3인분       보통        NaN   

       PC_NM                                            IMG_URL  \
0     5,000원  http://file.okdab.com/UserFiles/searching/reci...   
1     3,000원  http://file.okdab.com/UserFiles/searching/reci...   
2        NaN  http://file.okdab.com/UserFiles/searching/reci...   
3        NaN  http://file.okdab.com/UserFiles/searching/reci...   
4        NaN  http://file.okdab.com/UserFiles/searching/reci...   
..       ...                                                ...   
532   5,000원  http://file.okdab.com/recipe/14829104368610000...   
533   3,000원  http://file.okdab.com/recipe/14829144377250001...   
534   5,000원  http://file.okdab.com/recipe/14829900265530011...   
535   3,000원  http://file.okdab.com/recipe/14829933250580012...   
536  10,000원  http://file.okdab.com/recipe/14829957726840013...   

                                               DET_URL  
0    http://www.okdab.com/consumer/recipe/recipeVie...  
1    http://www.okdab.com/consumer/recipe/recipeVie...  
2    http://www.okdab.com/consumer/recipe/recipeVie...  
3    http://www.okdab.com/consumer/recipe/recipeVie...  
4    http://www.okdab.com/consumer/recipe/recipeVie...  
..                                                 ...  
532  http://www.okdab.com/consumer/recipe/recipeVie...  
533  http://www.okdab.com/consumer/recipe/recipeVie...  
534  http://www.okdab.com/consumer/recipe/recipeVie...  
535  http://www.okdab.com/consumer/recipe/recipeVie...  
536  http://www.okdab.com/consumer/recipe/recipeVie...  

[537 rows x 17 columns]

In [3]:
df = df[df['NATION_NM']=='한식']

In [4]:
jae = pd.read_csv('./datasets/ingre_utf8.csv')
jae

Unnamed: 0  ROW_NUM  RECIPE_ID  IRDNT_SN IRDNT_NM IRDNT_CPCTY  \
0              0        1          1         1        쌀          4컵   
1              1        2          1         2       안심        200g   
2              2        3          1         3      콩나물         20g   
3              3        4          1         4      청포묵        1/2모   
4              4        5          1         5      미나리         20g   
...          ...      ...        ...       ...      ...         ...   
6099        6099     3996        360      3996      깨소금          약간   
6100        6100     3997        360      3997        조         30g   
6101        6101     3998        360      3998        콩         30g   
6102        6102     3999        361      3999        무  2011-01-02   
6103        6103     4000        361      4000      미나리        10줄기   

      IRDNT_TY_CODE IRDNT_TY_NM  
0           3060001         주재료  
1           3060001         주재료  
2           3060001         주재료  
3           3060001         주재료  
4           3060001         주재료  
...             ...         ...  
6099        3060003          양념  
6100        3060002         부재료  
6101        3060002         부재료  
6102        3060001         주재료  
6103        3060002         부재료  

[6104 rows x 8 columns]

In [5]:
aa = jae['IRDNT_NM'].unique()

In [6]:
len(aa)

747

In [7]:
all = pd.merge(df, jae, on='RECIPE_ID')
all

Unnamed: 0_x  ROW_NUM_x  RECIPE_ID RECIPE_NM_KO  \
0                0          1          1        나물비빔밥   
1                0          1          1        나물비빔밥   
2                0          1          1        나물비빔밥   
3                0          1          1        나물비빔밥   
4                0          1          1        나물비빔밥   
...            ...        ...        ...          ...   
4234           536        537     195453      콩비지동그랑땡   
4235           536        537     195453      콩비지동그랑땡   
4236           536        537     195453      콩비지동그랑땡   
4237           536        537     195453      콩비지동그랑땡   
4238           536        537     195453      콩비지동그랑땡   

                                SUMRY  NATION_CODE NATION_NM  TY_CODE TY_NM  \
0     육수로 지은 밥에 야채를 듬뿍 넣은 영양만점 나물비빔밥!      3020001        한식  3010001     밥   
1     육수로 지은 밥에 야채를 듬뿍 넣은 영양만점 나물비빔밥!      3020001        한식  3010001     밥   
2     육수로 지은 밥에 야채를 듬뿍 넣은 영양만점 나물비빔밥!      3020001        한식  3010001     밥   
3     육수로 지은 밥에 야채를 듬뿍 넣은 영양만점 나물비빔밥!      3020001        한식  3010001     밥   
4     육수로 지은 밥에 야채를 듬뿍 넣은 영양만점 나물비빔밥!      3020001        한식  3010001     밥   
...                               ...          ...       ...      ...   ...   
4234     두부대신 콩비지를 넣어 만든 동그랑땡 맛도 좋아요!      3020001        한식  3010018    부침   
4235     두부대신 콩비지를 넣어 만든 동그랑땡 맛도 좋아요!      3020001        한식  3010018    부침   
4236     두부대신 콩비지를 넣어 만든 동그랑땡 맛도 좋아요!      3020001        한식  3010018    부침   
4237     두부대신 콩비지를 넣어 만든 동그랑땡 맛도 좋아요!      3020001        한식  3010018    부침   
4238     두부대신 콩비지를 넣어 만든 동그랑땡 맛도 좋아요!      3020001        한식  3010018    부침   

     COOKING_TIME  ...    PC_NM  \
0             60분  ...   5,000원   
1             60분  ...   5,000원   
2             60분  ...   5,000원   
3             60분  ...   5,000원   
4             60분  ...   5,000원   
...           ...  ...      ...   
4234          30분  ...  10,000원   
4235          30분  ...  10,000원   
4236          30분  ...  10,000원   
4237          30분  ...  10,000원   
4238          30분  ...  10,000원   

                                                IMG_URL  \
0     http://file.okdab.com/UserFiles/searching/reci...   
1     http://file.okdab.com/UserFiles/searching/reci...   
2     http://file.okdab.com/UserFiles/searching/reci...   
3     http://file.okdab.com/UserFiles/searching/reci...   
4     http://file.okdab.com/UserFiles/searching/reci...   
...                                                 ...   
4234  http://file.okdab.com/recipe/14829957726840013...   
4235  http://file.okdab.com/recipe/14829957726840013...   
4236  http://file.okdab.com/recipe/14829957726840013...   
4237  http://file.okdab.com/recipe/14829957726840013...   
4238  http://file.okdab.com/recipe/14829957726840013...   

                                                DET_URL Unnamed: 0_y  \
0     http://www.okdab.com/consumer/recipe/recipeVie...            0   
1     http://www.okdab.com/consumer/recipe/recipeVie...            1   
2     http://www.okdab.com/consumer/recipe/recipeVie...            2   
3     http://www.okdab.com/consumer/recipe/recipeVie...            3   
4     http://www.okdab.com/consumer/recipe/recipeVie...            4   
...                                                 ...          ...   
4234  http://www.okdab.com/consumer/recipe/recipeVie...         1099   
4235  http://www.okdab.com/consumer/recipe/recipeVie...         1100   
4236  http://www.okdab.com/consumer/recipe/recipeVie...         1101   
4237  http://www.okdab.com/consumer/recipe/recipeVie...         1102   
4238  http://www.okdab.com/consumer/recipe/recipeVie...         1103   

     ROW_NUM_y IRDNT_SN IRDNT_NM  IRDNT_CPCTY  IRDNT_TY_CODE  IRDNT_TY_NM  
0            1        1        쌀           4컵        3060001          주재료  
1            2        2       안심         200g        3060001          주재료  
2            3        3      콩나물          20g        3060001          주재료  
3            4        4      청포묵         1/2모        3060001          주재료

In [8]:
all = all[all['IRDNT_TY_NM']=='주재료']
all

Unnamed: 0_x  ROW_NUM_x  RECIPE_ID RECIPE_NM_KO  \
0                0          1          1        나물비빔밥   
1                0          1          1        나물비빔밥   
2                0          1          1        나물비빔밥   
3                0          1          1        나물비빔밥   
4                0          1          1        나물비빔밥   
...            ...        ...        ...          ...   
4234           536        537     195453      콩비지동그랑땡   
4235           536        537     195453      콩비지동그랑땡   
4236           536        537     195453      콩비지동그랑땡   
4237           536        537     195453      콩비지동그랑땡   
4238           536        537     195453      콩비지동그랑땡   

                                SUMRY  NATION_CODE NATION_NM  TY_CODE TY_NM  \
0     육수로 지은 밥에 야채를 듬뿍 넣은 영양만점 나물비빔밥!      3020001        한식  3010001     밥   
1     육수로 지은 밥에 야채를 듬뿍 넣은 영양만점 나물비빔밥!      3020001        한식  3010001     밥   
2     육수로 지은 밥에 야채를 듬뿍 넣은 영양만점 나물비빔밥!      3020001        한식  3010001     밥   
3     육수로 지은 밥에 야채를 듬뿍 넣은 영양만점 나물비빔밥!      3020001        한식  3010001     밥   
4     육수로 지은 밥에 야채를 듬뿍 넣은 영양만점 나물비빔밥!      3020001        한식  3010001     밥   
...                               ...          ...       ...      ...   ...   
4234     두부대신 콩비지를 넣어 만든 동그랑땡 맛도 좋아요!      3020001        한식  3010018    부침   
4235     두부대신 콩비지를 넣어 만든 동그랑땡 맛도 좋아요!      3020001        한식  3010018    부침   
4236     두부대신 콩비지를 넣어 만든 동그랑땡 맛도 좋아요!      3020001        한식  3010018    부침   
4237     두부대신 콩비지를 넣어 만든 동그랑땡 맛도 좋아요!      3020001        한식  3010018    부침   
4238     두부대신 콩비지를 넣어 만든 동그랑땡 맛도 좋아요!      3020001        한식  3010018    부침   

     COOKING_TIME  ...    PC_NM  \
0             60분  ...   5,000원   
1             60분  ...   5,000원   
2             60분  ...   5,000원   
3             60분  ...   5,000원   
4             60분  ...   5,000원   
...           ...  ...      ...   
4234          30분  ...  10,000원   
4235          30분  ...  10,000원   
4236          30분  ...  10,000원   
4237          30분  ...  10,000원   
4238          30분  ...  10,000원   

                                                IMG_URL  \
0     http://file.okdab.com/UserFiles/searching/reci...   
1     http://file.okdab.com/UserFiles/searching/reci...   
2     http://file.okdab.com/UserFiles/searching/reci...   
3     http://file.okdab.com/UserFiles/searching/reci...   
4     http://file.okdab.com/UserFiles/searching/reci...   
...                                                 ...   
4234  http://file.okdab.com/recipe/14829957726840013...   
4235  http://file.okdab.com/recipe/14829957726840013...   
4236  http://file.okdab.com/recipe/14829957726840013...   
4237  http://file.okdab.com/recipe/14829957726840013...   
4238  http://file.okdab.com/recipe/14829957726840013...   

                                                DET_URL Unnamed: 0_y  \
0     http://www.okdab.com/consumer/recipe/recipeVie...            0   
1     http://www.okdab.com/consumer/recipe/recipeVie...            1   
2     http://www.okdab.com/consumer/recipe/recipeVie...            2   
3     http://www.okdab.com/consumer/recipe/recipeVie...            3   
4     http://www.okdab.com/consumer/recipe/recipeVie...            4   
...                                                 ...          ...   
4234  http://www.okdab.com/consumer/recipe/recipeVie...         1099   
4235  http://www.okdab.com/consumer/recipe/recipeVie...         1100   
4236  http://www.okdab.com/consumer/recipe/recipeVie...         1101   
4237  http://www.okdab.com/consumer/recipe/recipeVie...         1102   
4238  http://www.okdab.com/consumer/recipe/recipeVie...         1103   

     ROW_NUM_y IRDNT_SN IRDNT_NM  IRDNT_CPCTY  IRDNT_TY_CODE  IRDNT_TY_NM  
0            1        1        쌀           4컵        3060001          주재료  
1            2        2       안심         200g        3060001          주재료  
2            3        3      콩나물          20g        3060001          주재료  
3            4        4      청포묵         1/2모        3060001          주재료

In [9]:
n = all['IRDNT_NM'].unique()
n

array(['쌀', '안심', '콩나물', '청포묵', '미나리', '숙주', '도라지', '고사리', '계란', '멥쌀',
       '찹쌀', '쇠고기', '청동호박', '흑임자', '밀가루', '감자', '냉면', '동치미국물', '쇠고기육수',
       '국수', '동치미무', '열무', '표고버섯', '오이', '부추', '오징어', '새우', '홍합', '김치',
       '만두피', '다시마', '두부', '모시조개', '북어', '무', '미역', '다진쇠고기', '불린미역', '생태',
       '물', '재첩', '실파', '느타리버섯', '양송이버섯', '물엿', '토마토케첩', '팥', '우엉', '잔멸치',
       '갈치', '닭', '칼국수면', '양배추잎', '밤', '양파', '마늘', '간장', '갈비', '당근',
       '돼기고기', '새우젓', '굴', '청국장', '순두부', '동태', '배추김치', '가래떡', '흰콩',
       '돼지고기', '배춧잎', '통북어', '멸치', '단무지', '해파리', '게맛살', '배추', '갓', '멸치젓',
       '풋배추', '홍고추', '총각무', '고추', '꽃게', '아구', '조기', '명란', '우거지', '된장',
       '팽이버섯', '낙지', '선지', '밥', '양', '소 잡뼈', '사골', '도가니', '양지머리', '당면',
       '깻잎', '콩', '연근', '조갯살', '냉이', '더덕', '꽈리고추', '마른오징어', '죽순', '애호박',
       '돼지갈비', '꽁치', '통도라지', '라면', '햄', '가는파', '청주', '굵은파', '생강', '식용유',
       '후춧가루', '도토리묵', '고등어', '가지', '마른새우', '골뱅이통조림', '배', '잣가루', '잣',
       '닭고기', '유부', '목이버섯', '양상추', '수삼', '프랑크소시지', '곱창', '쑥갓', '

In [10]:
len(n)

304

한식의 전체 재료 : 503

한식의 주재료 : 304

한식의 부재료 : 226

한식의 양념 : 167

전체 재료 : 6014